randomwallpaper@iflow.space/wallpaperController.js

```diff
@@ -252,54 +246,91 @@ var WallpaperController = class {
 	async _setBackground(path, callback) {
 		let monitorCount = Utils.Utils.getMonitorCount();
 		let background_setting = new Gio.Settings({ schema: "org.gnome.desktop.background" });
+		let screensaver_setting = new Gio.Settings({ schema: "org.gnome.desktop.screensaver" });
 		let wallpaperUri = "file://" + path;
 
-		try {
-			if (this._settings.get('multiple-displays', 'boolean') && await Utils.Utils.getHydraPaperAvailable()) {
-				// Needs a copy here
-				let hydraPaperCommand = [...Utils.Utils.getHydraPaperCommand()];
-
-				hydraPaperCommand.push('--cli');
-				hydraPaperCommand.push(path);
-
-				// Abuse history to fill missing images
-				for (let index = 0; index < monitorCount - 1; index++) {
-					let historyElement;
-					do {
-						historyElement = this._historyController.getRandom();
-					} while (this._historyController.history.length > monitorCount && hydraPaperCommand.includes(historyElement.path, 1))
-					// ensure different wallpaper for all displays if possible
-
-					hydraPaperCommand.push(historyElement.path);
-				}
+		let changeType = this._settings.get('change-type', 'enum');
+		// <value value='0' nick='Background' />
+		// <value value='1' nick='Lock Screen' />
+		// <value value='2' nick='Background and Lock Screen' />
+		// TODO: <value value='3' nick='Background and Lock Screen independently' />
 
-				try {
-					this._hydraPaperCancellable = new Gio.Cancellable();
+		if (changeType === 0 || changeType === 2) {
+			try {
+				if (this._settings.get('multiple-displays', 'boolean') && this._hydraPaper !== null && await this._hydraPaper.isAvailable()) {
+					let wallpaperArray = [path];
+
+					// Abuse history to fill missing images
+					for (let index = 0; index < monitorCount - 1; index++) {
+						let historyElement;
+						do {
+							historyElement = this._historyController.getRandom();
+						} while (this._historyController.history.length > monitorCount && wallpaperArray.includes(historyElement.path, 1))
+						// ensure different wallpaper for all displays if possible
+
+						wallpaperArray.push(historyElement.path);
+					}
```

Comment:
Only one monitor will get a new wallpaper for now?

Reply:
At this time yes. Random history wallpaper will fill missing displays.
Sadly a bit buried, in [`572329c` (#154)](https://github.com/ifl0w/RandomWallpaperGnome3/pull/154/commits/572329c5ffa5668742a375beb92afd985ec83d48) introduces requesting multiple new images.